presto-docs/src/main/sphinx/gcp/storage.rst

```diff
@@ -0,0 +1,216 @@
+Querying Google Cloud Storage
+=============================
+
+Preliminary Steps
+-----------------
+
+Ensure Access to GCS
+^^^^^^^^^^^^^^^^^^^^
+
+If your data is publicly available, you do not need to do anything here.
+However, in most cases data is not publicly available, and the Presto cluster needs to have access to it.
+This is typically achieved by creating a service account which has permissions to access your data.
+You can do this on the `service accounts page in GCP <https://console.cloud.google.com/projectselector2/iam-admin/serviceaccounts>`_.
+Once you create a service account, create a key for it and download the key in JSON format.
+
+Configure Cloud Storage connector
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Access to Cloud Storage data is possible thanks to `Hadoop Cloud Storage connector <https://cloud.google.com/dataproc/docs/concepts/connectors/cloud-storage>`_.
+It allows Presto (and other components using Hadoop) to access files in GCS via the ``gs://`` protocol.
+
+There are two ways to configure this connector:
+
+* Simplified Presto-specific configuration via Hive catalog properties (``hive.properties``)
+* Full configuration for the entire Hadoop ecosystem
+
+The simplified configuration method is useful if only Presto needs to access files in GCS, and is described in the `Hive Connector`_ section.
+If another Hadoop-based component (e.g. Hive metastore) also needs to access these files, it will be necessary to perform full configuration.
+
+.. _Full GCS configuration:
```

Comment:
These section is well written, but not appropriate as part of Presto's documentation. We don't have any other docs about setting up Hadoop components, and don't want to be in that business as it's complex and often vendor specific. There are at least three different guides on this:

* https://cloud.google.com/dataproc/docs/concepts/connectors/install-storage-connector
* https://docs.hortonworks.com/HDPDocuments/HDP3/HDP-3.1.0/bk_cloud-data-access/content/gcp-get-started.html
* https://www.cloudera.com/documentation/enterprise/latest/topics/admin_gcs_config.html

We also don't encourage using Hadoop config files with Presto. Normally, everything should be configured through Hive connector properties. If people need to fall back to Hadoop config files, we should address that by adding additional properties.

Reply:
Thanks, I'll replace the XML example with these links.